### DISCUSSION TITLE
False negative testing-library/no-unnecessary-act

### DISCUSSION THREAD

**Discussion by douglasjunior** [testing-library/eslint-plugin-testing-library#685] [Q&A]
In one test case we called a function within ACT that ESLint indicates is a testing-library function, but is not.

![image](https://user-images.githubusercontent.com/1512341/198692772-eca3d026-f671-46d9-b6cd-399317f01ab1.png)

And `dismissCallback` need to be wrapped with `act` because it dispatches a state update to close the AlertModal.

```tsx
  it('should not auto close modal with autoDismiss=false', async () => {
    const mockOnDismiss = jest.fn();

    let dismissCallback: undefined | (() => void);

    const tree = render(
      <AlertModal
        visible
        title="Title"
        description="Lorem ipsum is placeholder text commonly used in the graphic"
        primaryButtonName="Primary Action"
        secondaryButtonName="Secondary Action"
        autoDismiss={false}
        onPressSecondary={event => {
          dismissCallback = event.dismiss;
        }}
        onDismiss={mockOnDismiss}
      />,
    );

    const buttonInstance = tree.getByTestId('alert-modal-secondary-button');
    fireEvent.press(buttonInstance);

    expect(dismissCallback).toBeDefined();

    expect(mockOnDismiss).not.toBeCalled();

    act(() => {
      dismissCallback?.();
    });

    expect(mockOnDismiss).toHaveBeenCalled();
```

It's a false negative, right?

Versions: 

```
    "@testing-library/jest-native": "4.0.13",
    "@testing-library/react-hooks": "8.0.1",
    "@testing-library/react-native": "11.2.0",
    "@types/jest": "26.0.23",
    "@types/react-test-renderer": "17.0.1",
    "babel-jest": "26.6.3",
    "jest": "26.6.3",
    "react-test-renderer": "17.0.2"
...
    "eslint": "7.32.0",
    "eslint-plugin-testing-library": "5.6.4"
```

**Comment by Belco90** [comment]
Hi @douglasjunior!

Interesting issue… It's definitely a false negative, but I have no clue what could be causing it! Could you try something for me: remove the optional chaining operator from the `dismissCallback` call, and let me know if that makes a difference:

```js
act(() => {
  dismissCallback();
});
```

**Reply by douglasjunior** [reply]
Hi @Belco90 , thank you for the answer!

Removing optional chaining also removes the error:

![image](https://user-images.githubusercontent.com/1512341/199008695-d168489f-48a2-4cce-8cbb-b933d7b7a023.png)

But, even with `if` the error occurs again:

![image](https://user-images.githubusercontent.com/1512341/199008841-a5d22fd5-c722-406f-b590-df1f1f4ceeb0.png)

**Reply by Belco90** [reply]
Thanks for the extra context! This is definitely a bug in the plugin, so I'll transform it into an issue. Thanks for reporting it.
